builders/integrations/oracles/dia.md

```diff
@@ -0,0 +1,289 @@
+
+  constructor() {
+      oracleUpdater = msg.sender;
+  }
+
+  function setRandomValue(
+    uint256 _round,
+    string memory _randomness,
+    string memory _signature,
+    string memory _previousSignature
+  ) public {
+    require(msg.sender == oracleUpdater, "not a updater");
+    require(lastRound < _round, "old round");
+    lastRound = _round;
+    values[_round] = Random(_randomness, _signature, _previousSignature);
+  }
+
+  function getValue(uint256 _round) external view returns (Random memory) {
+    return values[_round];
+  }
+
+  function updateOracleUpdaterAddress(address newOracleUpdaterAddress)
+    public
+  {
+    require(msg.sender == oracleUpdater, "not a updater");
+    oracleUpdater = newOracleUpdaterAddress;
+    emit UpdaterAddressChange(newOracleUpdaterAddress);
+  }
+
+  function getRandomValueFromRound(uint256 _round)
+    external
+    view
+    returns (string memory)
+  {
+    return values[_round].randomness;
+  }
+
+  function getRandomValueFromRoundWithSignature(uint256 _round)
+    external
+    view
+    returns (Random memory)
+  {
+    return values[_round];
+  }
+
+    function getLastRound() public view returns (uint256) {
+    return lastRound;
+  }
+}
+```
+
+!!! note
+    DIA demo oracles are not intended for use in production environments. Developers can request a dedicated, production-ready randomness oracle. To start the request process, you can check out the [Request a Random Oracle](https://docs.diadata.org/products/randomness-oracle/request-a-random-oracle){target=_blank} documentation.
+
+### How to Use the DIA Randomness Oracle {: #how-to-use-the-dia-randomness-oracle }
+
+The steps for accessing a published random value are as follows:
+
+1. Access your randomness oracle smart contract on Moonbeam
+2. Call `getLastRound()`to obtain the ID of the latest published round. You can use the **Read Contract** functionality under the **Contract** tab of the contract on Moonscan to execute this call
+3. Call `getRandomValueFromRound(uint256 _round)` using the obtained round ID. Again, you can use Moonscan to quickly execute this call
+
+The response contains the randomness value.
+
+The signature can also be requested by calling `getRandomValueFromRoundWithSignature(uint256 _round)`, which returns a tuple containing the randomness value, the signature, and the previous signature.
+
+To learn how to deploy a randomness-consuming contract on Moonbeam, please refer to the [Deploying a #Randomness Consuming Smart Contract on #EVM chains with DIA xRandom #Oracle](https://youtu.be/BzN-tBgW-xs){target=_blank} video tutorial.
+
+# Learn More About DIA {: #learn-more-about-dia }
+
+- [Twitter](https://twitter.com/DIAdata_org){target=_blank}
+- [Discord](https://discord.gg/dia-dao){target=_blank}
+- [Website](https://diadata.org/){target=_blank}
+- [Docs](https://docs.diadata.org/){target=_blank}
+- [Explore data](https://www.diadata.org/app/){target=_blank}
+
```

Comment:
<img width="883" alt="Screenshot 2023-10-09 at 5 08 00 PM" src="https://github.com/moonbeam-foundation/moonbeam-docs/assets/5834262/88a01560-7730-45d3-9381-be85e2649d44">

This last section I would rename Resources and drop it down from a # title to a ## title. Or perhaps keep the name but add a  sentence or two of description. Right now it just feels a bit empty.

Reply:
Oh yeah didn't realize it was a # title... we don't use those besides the actual title of the page. good catch!